x-pack/elastic-agent/pkg/core/logger/logger.go

```diff
@@ -115,5 +115,24 @@ func makeInternalFileOutput() (zapcore.Core, error) {
 	}
 
 	encoder := zapcore.NewJSONEncoder(ecszap.ECSCompatibleEncoderConfig(logp.JSONEncoderConfig()))
-	return ecszap.WrapCore(zapcore.NewCore(encoder, rotator, zapcore.DebugLevel)), nil
+	// return ecszap.WrapCore(zapcore.NewCore(encoder, rotator, zapcore.DebugLevel)), nil
+	return ecszap.WrapCore(zapcore.NewCore(encoder, rotator, zapLevel(cfg.Level))), nil
+}
+
+func zapLevel(level logp.Level) zapcore.Level {
+	zapLevel := zapcore.DebugLevel
+	switch level {
+	case logp.DebugLevel:
+		zapLevel = zapcore.DebugLevel
+	case logp.InfoLevel:
+		zapLevel = zapcore.InfoLevel
+	case logp.WarnLevel:
+		zapLevel = zapcore.WarnLevel
+	case logp.ErrorLevel:
+		zapLevel = zapcore.ErrorLevel
+	case logp.CriticalLevel:
+		zapLevel = zapcore.ErrorLevel
+	}
+
+	return zapLevel
```

Comment:
Isn't there something that already does this in libbeat? Couldn't we change that to expose it, instead of implementing it here.